### PR TITLE
Getting Started adjustments, TLC...

### LINKED ISSUE #23731: Manual GCC/LLVM instruction update/TLC
## Reason
I see some issues with the current manual instructions set.

## Design
Correct the instructions. One glaring issue is instructing the user to gain a sane environment with:
```bash
bash --noprofile
```
No. That's not how you do it.

## Impact
Better instructions.

### PR DESCRIPTION
**Pull request by milljm**
Modify most of the install instructions for some TCL.

Generalize the Manual Install Instructions a bit to be more of a reference and less "copy/paste". The goal is to move ownership away from mooseframework.org and more towards each library's respected site.

Closes #23731

### CODE REVIEW

File: modules/doc/content/getting_started/installation/manual_gcc.md [hunk 1]
```diff
@@ -1,64 +1,40 @@
...
-cd $STACK_SRC
 curl -L -O http://mirrors.concertpass.com/gcc/releases/gcc-__GCC__/gcc-__GCC__.tar.gz
 tar -xf gcc-__GCC__.tar.gz -C .
 !package-end!
 
 Obtain GCC pre-reqs:
 
 !package! code max-height=450
-cd $STACK_SRC/gcc-__GCC__
+cd gcc-10.4.0
```
Comment (GiudGiud) [inline modules/doc/content/getting_started/installation/manual_gcc.md:17]: do you want to remove the variable? seems you are adding more text variables in this PR

File: modules/doc/content/getting_started/installation/manual_gcc.md [hunk 2]
```diff
@@ -1,64 +1,40 @@
 ## GCC
 
-We need a modern C++17 capable compiler. Our minimum requirements are: GCC [!package!minimum_gcc], Clang [!package!minimum_clang]. This section will focus on building a GCC [!package!gcc] compiler stack.
+We need a modern C++17 capable compiler. Our minimum requirements are: GCC [!package!minimum_gcc],
+Clang [!package!minimum_clang]. This section will focus on building a GCC [!package!gcc] compiler
+stack.
```
Comment (GiudGiud) [inline modules/doc/content/getting_started/installation/manual_gcc.md:5]:
```suggestion
stack. Please skip if you already have a GCC compiler of a sufficiently new version.
```

File: modules/doc/content/getting_started/installation/manual_installation_gcc.md [hunk 3]
```diff
@@ -1,58 +1,17 @@
...
-
-Thats it! Now you can either source this file manually each time you need to work on a MOOSE based
-application:
-
-```bash
-source /path/to/moose-environment.sh
-```
-
-Or you can permanently have it loaded each time you open a terminal by adding the above `source`
-command in your ~/.bash_profile (or ~/.bashrc which ever your system uses).
```
Comment (GiudGiud) [inline modules/doc/content/getting_started/installation/manual_installation_gcc.md:54]: where is this advice now?

File: modules/doc/content/getting_started/installation/manual_installation_llvm.md [hunk 4]
```diff
@@ -1,79 +1,25 @@
...
-Getting a proper LLVM/MPICH compiler built is substantially more difficult than our [GCC/MPICH](getting_started/installation/manual_installation_gcc.md) combination. We need to build LLVM using a *modern* GCC compiler (not just a minimal C++17 compliant GCC compiler). Also, the higher the version of GCC, does not always guarantee success.
-
-Furthermore, your millage will vary depending on the operating system. Building LLVM is a finicky process. If you are unsure as to why you are building an LLVM compiler, it might be best to build the [GCC/MPICH](getting_started/installation/manual_installation_gcc.md) compiler stack instead.
+!alert! note title=Are you sure!? Are you sure!? Are you sure!?
+Building an LLVM (Clang) compiler is substantially more difficult than GCC
 !alert-end!
 
-!include sqa/minimum_requirements.md
+Because we need MPICH to ultimately wrap to a Fortran compiler, it will be necessary to build GCC
+first, and then LLVM Clang. Therefor both compilers must be built *perfectly* (+you *really* need to
```
Comment (GiudGiud) [inline modules/doc/content/getting_started/installation/manual_installation_llvm.md:8]:
```suggestion
first, and then LLVM Clang. Therefore both compilers must be built *perfectly* (+you *really* need to
```

**Review by GiudGiud — CHANGES_REQUESTED** [review]
Somebody more expert will give a better review

File: modules/doc/content/getting_started/installation/offline_installation.md [hunk 5]
```diff
@@ -1,46 +1,37 @@
...
-MVAPICH is regarded as the better wrapper for HPC clusters. However, it is also the most difficult to build properly for that clusters specific hardware. While you may be able to *build* it easily enough, you may not have built it *properly* to optimally make use of said cluster's hardware.
+MVAPICH is regarded as the better wrapper for HPC clusters. However, it is also the most difficult
+to build properly for that clusters specific hardware. While you may be able to *build* it easily
+enough, you may not have built it *properly* to optimally make use of said cluster's hardware.
 
-## Prepare PETSc, libMesh, and MOOSE for copying
+## Prepare Directory
 
-These three libraries can all be obtained from one repository (the moose repository). First, create a top-level directory named 'offline', which will ultimately contain everything we need to transfer over to your target machine.
+These three libraries can all be obtained from one repository (the moose repository). First, create
```
[the remark this answers is not in the document]
Comment (milljm) [reply modules/doc/content/getting_started/installation/offline_installation.md:32]: Yeah... man I always disliked these instructions. They will always be wrong (the links to those PETSc contribs are always changing).

I almost think we should differ users to Containerized solutions instead. Like Docker. I'll have to think on this. I'll make the changes, but I think in the future this page will be replaced by a containerized solution.

File: modules/doc/content/getting_started/installation/hpc_install_moose.md [hunk 6]
```diff
@@ -1,42 +1,45 @@
...
 ```bash
-source $HOME/.moose_profile
+export CC=mpicc CXX=mpicxx FC=mpif90 F90=mpif90 F77=mpif77
 ```
 
-By sourcing the above file, you are now ready to begin MOOSE-based development.
+!template load file=installation/clone_moose.md.template PATH=~/projects
 
-!alert note title=+Remember to source the profile!+
-You will need to perform the above (`source $HOME/.moose_profile`) for every new terminal session for which you perform work with MOOSE. If you want this to be automatic, add the above to your `~/.bash_profile` (or `~/.bashrc` or, which ever profile you use on your system)
```
[the remark this answers is not in the document]
Comment (milljm) [reply modules/doc/content/getting_started/installation/hpc_install_moose.md:40]: We are instructing folks to use their HPC clusters recommended methods to aquire an MPI environment (like INL's `module load use.moose moose-dev`) So there is no need to provide these anymore.

File: modules/doc/config.yml [hunk 7]
```diff
@@ -41,7 +41,7 @@ Extensions:
         google-cse: 000781241768298771085:pedtnlxdkx8
         menu:
             Getting Started:
-                Installing MOOSE: getting_started/installation/index.md
+                Building MOOSE: getting_started/installation/index.md
```
[the remark this answers is not in the document]
Comment (cticenhour) [reply modules/doc/config.yml:44]: I disagree with saying we're telling folks how to create their own here. The vast majority of these discussions are regarding installation of system dependencies required for MOOSE usage/development, and then building MOOSE (which is installing for a user who will only be using the built code).  

I think "Installing MOOSE" achieves the clarity we're looking for with those who are seeing the site for the first time.

File: modules/doc/content/getting_started/installation/manual_installation_gcc.md (hunk 3 above)
Comment (milljm) [reply modules/doc/content/getting_started/installation/manual_installation_gcc.md:54]: @cticenhour @loganharbour Do you have any suggestions on this?

Comment (loganharbour) [reply modules/doc/content/getting_started/installation/manual_installation_gcc.md:54]: Nuke it. We put a lot of effort into providing a stable stack for various kinds of users. And that will get even better when we get containers out.

These kinds of instructions should only be used now by people that get paid to manage software stacks. And those people shouldn't need these kinds of instructions.

Comment (cticenhour) [reply modules/doc/content/getting_started/installation/manual_installation_gcc.md:54]: I agree with Logan's take. Let's do away with it.

File: modules/doc/content/getting_started/installation/offline_installation.md [hunk 8]
```diff
@@ -231,16 +244,23 @@ cd $MOOSE_DIR/test
...
 ```bash
 export PATH=/some/path/to/MPI/bin:/some/path/to/GCC/bin:$PATH <--EXAMPLE, CHANGE ME
 export CC=mpicc CXX=mpicxx FC=mpif90 F90=mpif90 F77=mpif77
 export MOOSE_DIR=$HOME/offline/moose
 ```
 
-You will need to perform the above, each and every time you log into the target machine and wish to perform development on your application. To that end, creating a special profile you can source in one command is recommended.
+You will need to perform the above, each and every time you log into the target machine and wish to
+perform development on your application. To that end, creating a special profile you can source in
+one command is recommended.
```
[the remark this answers is not in the document]
Comment (milljm) [reply modules/doc/content/getting_started/installation/offline_installation.md:272]: I always seem to miss a change  request... Yeah I will add a `!alert tip` here

[the remark this answers is not in the document]
Comment (GiudGiud) [reply modules/doc/content/getting_started/installation/offline_installation.md:273]: I was thinking a snippet like:
```
export CC=....
```
and how to export that to a bash script people can run

File: modules/doc/content/getting_started/installation/test_moose.md [hunk 9]
```diff
@@ -1,9 +1,9 @@
-## Compile and Test MOOSE
+## Build and Test MOOSE
```
Comment (GiudGiud) [inline modules/doc/content/getting_started/installation/test_moose.md:1]: SHould you use the template you created ?

modules/doc/content/getting_started/installation/test_moose.md.template

File: modules/doc/content/getting_started/installation/offline_installation.md [hunk 10]
```diff
@@ -1,52 +1,50 @@
...
-It is highly likely the target machine in which you will be operating on will already contain an acceptable version of GCC. Please contact your system administrator for instructions on how to use that systems compiler. If not, you will need to head on over to GNU's [GCC](https://gcc.gnu.org/) site, and follow their instructions on building a GCC compiler at or beyond the above listed minimum version.
+- [MPICH](https://www.mpich.org/)
 
-## LLVM (optional)
+- [OpenMPI](https://www.open-mpi.org/)
 
-While arguably a [better compiler](https://opensource.apple.com/source/clang/clang-23/clang/tools/clang/www/comparison.html) over traditional GCC, it is more difficult to build. Nevertheless, if you wish to make the attempt, head on over to [llvm.org](http://llvm.org/), to begin.
+!alert note title=Personal Air-Gapped Workstation
+Procuring the above on a personal air-gapped workstation will need to be your responsibility. We
+cannot instruct a means for bypassing such security.
```
Comment (GiudGiud) [inline modules/doc/content/getting_started/installation/offline_installation.md:35]: not sure what you mean here.
bypass the security of air-gapping? why are we "bypassing" ?

File: modules/doc/content/getting_started/installation/test_moose.md (hunk 9 above)
Comment (milljm) [reply modules/doc/content/getting_started/installation/test_moose.md:1]: We can't... If I am thinking correctly when I say we may not even need `test_moose.md.template` anymore (I can't remember. But I know when I tried to use the other templates we fail; clone_moose.md.template and the other one).

I ran into template recursion issues when attempting to use templates for our basic instructions. So these continue to need not use templates (keep the originals). But as far as our Advanced instructions, the templates work.

Sorta makes me wonder if its even worth templating for just two pages.

File: modules/doc/content/getting_started/installation/offline_installation.md (hunk 10 above)
Comment (milljm) [reply modules/doc/content/getting_started/installation/offline_installation.md:35]: I don't want to instruct people how to create an SSH reverse tunnel. Or how to use/bring in a thumb stick into an obviously required air-gapped secure network...

They will need to figure out how to install these prerequisites on their own. We can't help them `apt-get install`, or `xcode-select`, or use App Store, etc, behind a non-internet accessing computer. This statement basically says: Contact your help desk.

File: modules/doc/content/getting_started/installation/manual_mpich.md.template [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+Check and see if you already have an MPI wrapper available on your machine. One simple way of doing
+so, is to perform a `which` on several MPI wrapper binaries:
+
+```bash
+which mpicc mpicxx mpif90
+```
+
+If the above command returns with paths to their respected binaries, you will want to work with your
+system package manager to remove or suppress it.
```
[the remark this answers is not in the document]
Comment (milljm) [reply modules/doc/content/getting_started/installation/manual_mpich.md.template:11]: Yeah, except the first clause:

```bash
The following instructions are for those who wish to build their own custom GCC compiler stack.
```

When you do this, you need to build an MPI wrapper based on the new compiler

**Review by GiudGiud — DISMISSED** [review]

File: modules/doc/content/getting_started/installation/offline_installation.md (hunk 10 above)
Comment (milljm) [reply modules/doc/content/getting_started/installation/offline_installation.md:43]: I can reword it a bit? Maybe something less security-like?

```bash
!alert note title=Personal Air-Gapped Workstation
Procuring the above on a machine with no network access is beyond the scope of this document. Please work with your system administrator on satisfying the above prerequisites. 
```

Comment (GiudGiud) [reply modules/doc/content/getting_started/installation/offline_installation.md:43]: That works

File: modules/doc/content/getting_started/installation/offline_installation.md [hunk 12]
```diff
@@ -39,8 +39,8 @@ If you are operating on a personal Linux workstation, the easiest way to obtain
 via the same tool used to obtain a developers environment in the Prerequisites section.
 
 !alert note title=Personal Air-Gapped Workstation
-Procuring the above on a personal air-gapped workstation will need to be your responsibility. We
-cannot instruct a means for bypassing such security.
+Procuring the above on a machine with no network access is beyond the scope of this document. Please
```
Comment (GiudGiud) [inline modules/doc/content/getting_started/installation/offline_installation.md:42]:
```suggestion
Procuring the above on a workstation with no network access is beyond the scope of this document. Please
```

HPC cluster is within right?

File: modules/doc/content/getting_started/installation/offline_installation.md [hunk 13]
```diff
@@ -213,12 +214,13 @@ cd ~/offline/moose/test
...
 
-However, most tests should pass. If they don't, or you see `MAX FAILURES`, thats a problem! And it
-needs to be addressed before continuing. Please supply a report of the actual failure (scroll up a
-ways). For example the following does not help us (created with `./run_tests -i always_bad`):
+You will also see SKIPPED tests. This is normal as some tests are specific to Macintosh, some for
+Linux. Or some other constraint your machine does not satisfy.
+
+Most tests should pass. If they don't, or you see `MAX FAILURES`, thats a problem! And it needs to
+be addressed before continuing. Please supply a report of the actual failure (scroll up a ways). For
+example the following explains little (created with `./run_tests -i always_bad`):
```
Comment (GiudGiud) [inline modules/doc/content/getting_started/installation/offline_installation.md:223]:
```suggestion
example the following snippet does not give the full picture (created with `./run_tests -i always_bad`):
```

File: modules/doc/content/getting_started/installation/offline_installation.md [hunk 14]
```diff
@@ -230,7 +232,7 @@ Ran 2 tests in 0.2 seconds. Average test time 0.0 seconds, maximum test time 0.0
 0 passed, 0 skipped, 0 pending, 2 FAILED
 ```
 
-You need to scroll up, and find/report the actual error:
+Instead, you need to scroll up and report the actual error:
```
Comment (GiudGiud) [inline modules/doc/content/getting_started/installation/offline_installation.md:235]: thanks for including this!

File: modules/doc/content/getting_started/installation/offline_installation.md [hunk 15]
```diff
@@ -253,19 +255,20 @@ tests/test_harness.always_bad .................................. FAILED (CODE 1)
...
-cluster, or personal workstation before you can perform any application development.
+cluster or personal workstation, before you can perform any application development.
 
-For this reason you may wish to add the above export commands to your shell startup profile. This is
-different for each shell type (and there are a lot). You'll want to read up on how to do this for
-Macintosh (usually ZSH: `~/.zshrc`), or Linux (usually BASH: `~/.bashrc`) respectfully.
+For this reason you may wish to add the above export commands to your shell startup profile so that
+these variables are always set. Achieving this is different for each shell type (and there are a
+lot). You'll want to read up on how to do this for Macintosh (usually ZSH: `~/.zshrc`), or Linux
+(usually BASH: `~/.bashrc`) respectfully.
```
Comment (GiudGiud) [inline modules/doc/content/getting_started/installation/offline_installation.md:271]:
```suggestion
(usually BASH: `~/.bashrc`) respectively.
```

File: modules/doc/content/getting_started/installation/test_moose.md (hunk 9 above)
Comment (milljm) [reply modules/doc/content/getting_started/installation/test_moose.md:1]: now that I am staring at the pages, yes... we should use templates!

Comment (milljm) [reply modules/doc/content/getting_started/installation/test_moose.md:1]: What urks me... is that the page renders in the end, just fine.

Comment (GiudGiud) [reply modules/doc/content/getting_started/installation/test_moose.md:1]: oh well. It's not the biggest re-use, you're removing a lot of those re-uses here anyway

File: modules/doc/content/getting_started/installation/offline_installation.md (hunk 14 above)
Comment (milljm) [reply modules/doc/content/getting_started/installation/offline_installation.md]: Made this into a template. 👍

File: modules/doc/content/getting_started/installation/offline_installation.md (hunk 12 above)
Comment (milljm) [reply modules/doc/content/getting_started/installation/offline_installation.md]: I am not sure I understand the question.

HPC is listed on another bullet point. I tried to separate them as best I could:

#### Air-Gapped HPC Clusters

and

#### Personal Workstation

The idea was not to make these too large, or use ## to include them in the TOC (top right table of contents). But make them large enough to separate them as sections.

Comment (milljm) [reply modules/doc/content/getting_started/installation/offline_installation.md]: I thought about indenting them (like we here for each section: https://mooseframework.inl.gov/help/troubleshooting.html) But we quickly run out of space :(

Sometimes I wish we would do away with the left margin!!!

File: modules/doc/content/getting_started/installation/test_moose.md (hunk 9 above)
Comment (milljm) [reply modules/doc/content/getting_started/installation/test_moose.md]: @cticenhour  helped me figure out the issue :) I was doing it wrong! Of course.

So now, moar templates incoming!

**Review by GiudGiud — DISMISSED** [review]

**Comment by milljm** [comment]
This is ready, but need to hold off, until I get the failing application PR ready.

**Review by GiudGiud — DISMISSED** [review]

**Comment by milljm** [comment]
This PR breaks Bison and Griffin documentation. For which I have a PR ready to go for them. Please merge when ready!

**Comment by milljm** [comment]
ok ok ok ok... i'll stop. done for the weekend.

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**